lib/fluent/plugin/in_systemd.rb

```diff
@@ -29,6 +29,7 @@ def start
       @running = true
       pos_writer.start
       @thread = Thread.new(&method(:run))
+      @thread.abort_on_exception = true
```

Comment:
As an external view, there is a race here, since the thread can start and the `abort_on_exception` flag is set concurrently, which means `#run` could exit silently if it happens to start executing before the flag is set. One fix would be to have instead `Thread.current.abort_on_exception = true` inside `#run` as the first statement.